muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/ChatRoomSyncScheduler.java

```diff
@@ -83,59 +106,94 @@ public void updateChatRooms() {
-        if (chatRoomEntity == null) {
-          logger.log(Level.INFO, "Syncing chat workspace " + enabledWorkspace);
-          SchoolDataIdentifier identifier = SchoolDataIdentifier.fromId(enabledWorkspace);
-          if (identifier == null) {
-            logger.log(Level.WARNING, "Invalid workspace identifier " + enabledWorkspace + ", skipping...");
-            continue;
-          }
-          WorkspaceEntity workspaceEntity = workspaceEntityController.findWorkspaceByIdentifier(identifier);
-          if (workspaceEntity == null) {
-            logger.log(Level.WARNING, "No workspace entity found for identifier " + enabledWorkspace + ", skipping...");
-            continue;
-          }
-
-          Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
-          chatRoomEntity = new MUCRoomEntity(enabledWorkspace, workspace.getName(), workspace.getDescription());
-          client.createChatRoom(chatRoomEntity);
-
-          List<WorkspaceUser> workspaceUsers = workspaceController.listWorkspaceStudents(workspaceEntity);
-          List<WorkspaceUser> workspaceStaffs = workspaceController.listWorkspaceStaffMembers(workspaceEntity);
-
-          for (WorkspaceUser workspaceStaff : workspaceStaffs) {
-
-            SchoolDataIdentifier memberIdentifier = workspaceStaff.getUserIdentifier();
-
-            if (enabledUsers.contains(memberIdentifier.toId())) {
-
-              client.addAdmin(enabledWorkspace, memberIdentifier.toId());
-
-            }
-          }
-
-          for (WorkspaceUser workspaceUser : workspaceUsers) {
-
-            SchoolDataIdentifier memberIdentifier = workspaceUser.getUserIdentifier();
-
-            if (enabledUsers.contains(memberIdentifier.toId())) {
-
-              client.addMember(enabledWorkspace, memberIdentifier.toId());
-
+    
+    Iterator<SearchProvider> searchProviderIterator = searchProviders.iterator();
+    if (searchProviderIterator.hasNext()) {
+      SearchProvider searchProvider = searchProviderIterator.next();
+      SearchResult searchResult = null;
+      
+      List<SchoolDataIdentifier> curriculumIdentifiers = null;
+      if (curriculums != null) {
+        curriculumIdentifiers = new ArrayList<>(curriculums.size());
+        for (Curriculum curriculum : curriculums) {
+          SchoolDataIdentifier curriculumIdentifier = SchoolDataIdentifier.fromId(curriculum.getIdentifier().toString());
+          if (curriculumIdentifier != null && !"OPS2005".equals(curriculum.getName())) {
+            curriculumIdentifiers.add(curriculumIdentifier);
+          } 
+        }
+      }
+      
+      String searchString = null;
+      boolean includeUnpublished = true;
+	  int firstResult = 0;
+  	  int maxResults = 100;
+	  List<Sort> sorts = null;
+	  searchResult = searchProvider.searchWorkspaces(null, null, null, null, curriculumIdentifiers, searchString, null, null, includeUnpublished, firstResult, maxResults, sorts);          
```

Comment:
Does this just go through the same 100 workspaces every 24 hours?

Might be better to do some utility that would be launched manually and which then goes through all workpaces to setup their chat rooms accordingly. For new workspaces, their chat rooms should probably be created when the workspace itself is created so that there wouldn't even be need for background maintenance.